kong/runloop/balancer/round_robin.lua

```diff
@@ -56,11 +55,15 @@ function roundrobin_algorithm:afterHostUpdate()
   end
 
   -- add all addresses to the wheel
-  for _, targets in ipairs(targets) do
-    for _, address in ipairs(targets.addresses) do
+  local new_wheel = {}
+  local idx = 1
+
+  for _, target in ipairs(targets) do
+    for _, address in ipairs(target.addresses) do
       local address_points = address.weight / divisor
       for _ = 1, address_points do
-        new_wheel[#new_wheel + 1] = address
+        new_wheel[idx] = address
+        idx = idx + 1
```

Comment:
`new_wheel[idx] = address; idx = idx + 1` seems the same to `table.insert(new_wheel, address)`, then we need not to define idx.

Reply:
It is a performance optimization. See: https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#writing-performant-code

Reply:
cool!